File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/SaveProjectUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.entity.ExportQuality
import com.izzatismail.midtrim.domain.entity.ProjectInfo
import com.izzatismail.midtrim.domain.entity.SourceVideoInfo
import com.izzatismail.midtrim.domain.error.ProjectError
import com.izzatismail.midtrim.domain.repository.ProjectRepository

class SaveProjectUseCase(
    private val repository: ProjectRepository
) {
    suspend fun execute(project: ProjectInfo, sourceVideos: List<SourceVideoInfo>) {
        if (project.name.isBlank()) {
            throw ProjectError.EmptyName
        }
        val trimmed = project.copy(name = project.name.trim())
        repository.save(trimmed, sourceVideos)
    }
}

File: ios/MidTrim/MidTrim/Domain/UseCases/SaveProjectUseCase.swift
import Foundation

struct SaveProjectUseCase {
    private let repository: ProjectRepositoryProtocol

    init(repository: ProjectRepositoryProtocol) {
        self.repository = repository
    }

    func execute(project: ProjectInfo, sourceVideos: [SourceVideoInfo]) async throws {
        guard !project.name.trimmingCharacters(in: .whitespaces).isEmpty else {
            throw ProjectError.emptyName
        }
        let trimmed = ProjectInfo(
            id: project.id, name: project.name.trimmingCharacters(in: .whitespaces),
            trimDuration: project.trimDuration, wasCustomDuration: project.wasCustomDuration,
            outputVideoURI: project.outputVideoURI, thumbnailURI: project.thumbnailURI,
            exportQualityTier: project.exportQualityTier, mergedDuration: project.mergedDuration,
            videoCount: project.videoCount, createdAt: project.createdAt, updatedAt: project.updatedAt
        )
        try await repository.save(project: trimmed, sourceVideos: sourceVideos)
    }
}
